### ISSUE TITLE
⌨ Accessibility Issue: Difficult to use radio buttons/ checkboxes

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#5492]
## ❌ Current behaviour (issue)
The edit fields for the month, year and day of the date display as if they are group combo boxes and edit boxes. The spin buttons for selecting a start date are difficult to use. If a users tabs to the start month, for example, they cannot simply arrow down to the month  want, and then tab to the next field. They have to arrow down to what they want, and then press enter. 
### How to reproduce
Open any screen reader in Windows and in Edge  or Chrome
Navigate to any date field on the platform (e,.g. Start and end date fields in the [Work Experience form](https://uat-talentcloud.tbs-sct.gc.ca/en/users/528357e6-0ff7-4809-8f90-0bca8198aaf7/profile/experiences/work/create), Closing date field on the [Edit pool advertisement](https://uat-talentcloud.tbs-sct.gc.ca/en/admin/pools/68c75ffd-8b24-4e6a-a6b9-8c0a2ed0b6b3/edit) page)
Try to enter a date using the edit options.
Try to enter a date using the spin boxes. 

### Potential WCAG Criteria Fable Users suggested for consideration: 
1.3.1 Info and Relationships, 
2.1.1 Keyboard,
2.1.2 No Keyboard Trap 
3.2.2 On Input
### Fable Video Example
[Date Field Challenges ](https://app.makeitfable.com/share/63c9ed0bd3499e0081ebf762)
## ✔ Expected behaviour (fix)
It is clear to screen reader users what feature to use to select a date. 
### Acceptance Criteria
A set of assumptions which, when tested, verify that the accessibility requirement was met.
- [ ] The edit fields for the month, year and day of the date do not display as if they are group combo boxes and edit boxes.
- [ ]  It is clear to screen reader users what feature to use to select a date. 
- [ ]  It should be specified in the ARIA label if the date selection feature is a picker, slider, or spin box.

**Comment by petertgiles** [comment]
Thanks for this one.  This is a bit sticky because we're using just a plain input element with "type=date" and the browser handles everything else.  The browser might show a calendar style picker on a desktop or a full screen spinner on iPhone.  We can't really customize this much unless we want to rebuild a date picker for ourselves.  

A similar discussion about the language of the date picker:  https://github.com/GCTC-NTGC/gc-digital-talent/issues/4314#issuecomment-1297482795

I know there have been discussions about making a new datepicker.  That would be a whole new feature that desfam would need to design for us.  Short of re-implementing the date picker are there any concrete actions the devs could take to improve this ?

**Comment by esizer** [comment]
Has this been solved by the new datepicker?

**Comment by RM-1978** [comment]
Yep @esizer, these are the tickets that underline the need for the new date picker.
